deepquest/utils/utils.py

```diff
@@ -32,16 +32,17 @@ def compare_params(params_new, params_old, ignore=None):
     """
     stop_flag = False
     for key in params_old:
-        if key not in (params_new or ignore):
-            logger.info(
-                'New config does not contain ' + key)
-            stop_flag = True
-        elif params_new[key] != params_old[key] and key not in ignore:
-            logger.info('Previous model has ' + key + ': ' +
-                        str(params_new[key]) + ' but this model has ' + key + ': ' + str(params_old[key]))
-            stop_flag = True
+        if key not in ignore:
+            if key not in params_new:
+                logger.info(
+                    'New config does not contain ' + key)
+                stop_flag = True
+            elif params_new[key] != params_old[key]:
+                logger.info('New model has ' + key + ': ' +
+                            str(params_new[key]) + ' but previous model has ' + key + ': ' + str(params_old[key]))
+                stop_flag = True
```

Comment:
@davidwilby, I don't think I've seen this function before, what is it that `changes2dict()`, `setparameters()` and `add_dependent_params()` don't already cover?
There are now many functions to deal with the parameters, and I'm struggling to get my head around what this handling mechanism has become.

Reply:
@fredblain When resuming a model, we need to check that the parameters that are specified by the user are the same as those used in the model training previously. This function (was previously just in train, though now it has been moved out into a function) checks the two parameters dictionaries for differences.

Reply:
I'm still not sure this is relevant, but let's keep as is for now as focus should be given to critical issues that are blocking the release.